common/motion-dashboard.cpp

```diff
@@ -0,0 +1,146 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2024 Intel Corporation. All Rights Reserved.
+
+#include "motion-dashboard.h"
+
+using namespace rs2;
+using namespace rsutils::string;
+
+motion_dashboard::motion_dashboard( std::string name )
+    : stream_dashboard( name, 30 )
+    , last_time( glfwGetTime() )
+    , x_value( 0 )
+    , y_value( 0 )
+    , z_value( 0 )
+    , n_value( 0 )
+{
+    frame_rate = MAX_FRAME_RATE;
+    clear( true );
+}
+
+void motion_dashboard::draw( ux_window & win, rect r )
+{
+    if( accel_params[curr_accel_param_position] == "X" )
+    {
+        auto x_hist = read_shared_data< std::deque< float > >( [&]() { return x_history; } );
+        for( int i = 0; i < x_hist.size(); i++ )
+        {
+            add_point( (float)i, (float)x_hist[i] );
+        }
+    }
+
+    if( accel_params[curr_accel_param_position] == "Y" )
+    {
+        auto y_hist = read_shared_data< std::deque< float > >( [&]() { return y_history; } );
+        for( int i = 0; i < y_hist.size(); i++ )
+        {
+            add_point( (float)i, (float)y_hist[i] );
+        }
+    }
+
+    if( accel_params[curr_accel_param_position] == "Z" )
+    {
+        auto z_hist = read_shared_data< std::deque< float > >( [&]() { return z_history; } );
+        for( int i = 0; i < z_hist.size(); i++ )
+        {
+            add_point( (float)i, (float)z_hist[i] );
+        }
+    }
+
+    if( accel_params[curr_accel_param_position] == "N" )
+    {
+        auto n_hist = read_shared_data< std::deque< float > >( [&]() { return n_history; } );
+        for( int i = 0; i < n_hist.size(); i++ )
+        {
+            add_point( (float)i, (float)n_hist[i] );
+        }
+    }
+    r.h -= ImGui::GetTextLineHeightWithSpacing() + 10;
+    draw_dashboard( win, r );
+
+    ImGui::SetCursorPosX( ImGui::GetCursorPosX() + ImGui::GetTextLineHeightWithSpacing() * 2 );
+    show_radiobuttons();
+
+    ImGui::SameLine();
+
+    show_data_rate_slider();
+}
+
+int motion_dashboard::get_height() const
+{
+    return (int)( 10 * ImGui::GetTextLineHeight() + ImGui::GetTextLineHeightWithSpacing() );
+}
+
+void motion_dashboard::clear( bool full )
+{
+    write_shared_data(
+        [&]()
+        {
+            if( full )
```

Comment:
If full is `false` we do nothing? When is it called with `false` and when with `true`?

Reply:
In the parent `stream_dashboard` class we have:
 `virtual void clear(bool full = false) {}`
 I found one case when we call with `full=true` value when a user wants clear logs and dashboard by writing `clear` command.

Reply:
What about closing or minimizing the dashboard? We don't want to collect/plot data when it is not shown

Reply:
You are right, we are not collecting data when the dashboard is minimized.
[Here line code](https://github.com/IntelRealSense/librealsense/pull/12671/files#diff-1a13fcecaf2babec10810470651a8e4f661854eab1694a508d474ede98e3f79bR734) that prevent dashboard work when it not open (fully open: we have smooth open animation)

Reply:
We are still collecting when minimized, just not drawing it. To stop collecting we should call stop/start on the dashboard and set a parameter. This enhancement can be done in next PR.